### PR TITLE
Todomvc-Aurelia 0.1.0

### PR DESCRIPTION
**Pull request by mhoyer**
This is an attempt to create the TodoMVC app with Aurelia.

## Pros

- fully TDD based development => 70 specs for release 0.1.0
- manually verified [TodoMVC App Specification](https://github.com/tastejs/todomvc/blob/master/app-spec.md) requirements
- development followed mainly the approaches of 
  - http://aurelia.io/get-started.html 
  - https://github.com/aurelia/skeleton-navigation
  - https://github.com/SAAirey/aurelia-skeleton-navigation-bundling

## "Cons"

Some things that still need to be discussed/considered:

- The original TodoMVC browser-test suite is currently not passing (at least on my Win7x64 machine).
  - See https://github.com/tastejs/todomvc/issues/1147
  - In turn, I see the same error when trying to run the reference implementation (backbone example) from latest todomvc repository. 
- The bundling and export of dependencies and actual application code doesn't feel mature (but it's automated through `gulp build`). 
  - How to provide an instant view for the user without flickering on page load? This is a somehow fundamental issue as SystemJS is recommended module loader for Auriala, which implies asynchronous loading of all dependencies and the app (incl. views) itself.
- Aurelia is still in pre-beta state. So changes in the API need to be applied for this sample application - what I'll try to do from time to time. ;-)

### CODE REVIEW

File: examples/aurelia/README.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+### Support
+
+- [Twitter](http://twitter.com/pixelplastic)
+- [Gitter](https://gitter.im/Aurelia/Discuss)
+
+*Let us [know](https://github.com/tastejs/todomvc/issues) if you discover anything worth sharing.*
+
+## Implementation
+
+- following the getting started tutorial: http://aurelia.io/get-started.html
```
Comment (sindresorhus) [inline examples/aurelia/README.md]: Capitalize the first character

File: examples/aurelia/README.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+- attempt to go TDD where possible
+
+### Changelog
+
+#### 0.1.0
+
+- finished functional implementation to fulfill [TodoMVC Application Specification](https://github.com/tastejs/todomvc/blob/master/app-spec.md)
+- includes bundling (jspm) and exporting (for easy sync with [todomvc/examples/aurelia](https://github.com/tastejs/todomvc/tree/master/examples/aurelia))
```
Comment (sindresorhus) [inline examples/aurelia/README.md]: Capitalize the first character

File: examples/aurelia/config.js [hunk 3]
```diff
@@ -0,0 +1,383 @@
+System.config({
+  "paths": {
```
Comment (sindresorhus) [inline examples/aurelia/config.js]: Don't quote keys and use single-quotes. This isn't JSON.

File: examples/aurelia/css/app.css [hunk 4]
```diff
@@ -0,0 +1 @@
+/* all comes from todomvc-app-css package */
```
Comment (sindresorhus) [inline examples/aurelia/css/app.css]: This can be removed.

File: examples/aurelia/dist/todos.html [hunk 5]
```diff
@@ -0,0 +1,48 @@
+<template>
+  <import from="./behaviors/focus"></import>
```
Comment (sindresorhus) [inline examples/aurelia/dist/todos.html]: `./` is moot.

**Comment by sindresorhus** [comment]
Found a couple of bugs:

- The 'All' filter should be active on load. (the one at the bottom)
- The app is missing the left "learn bar". See the other apps.

**Comment by mhoyer** [comment]
@sindresorhus thanks for good feedback. I'll apply your findings possibly later this day.

**Comment by mhoyer** [comment]
@sindresorhus I just added a squashed commit containing the fixes for the issues you pointed out. Were you able to let the browser-test suite run against the Aurelia example on your side? Or anyone else?

**Comment by mhoyer** [comment]
What's the status on this PR? Do you need any more support from my side?

**Comment by sindresorhus** [comment]
I'm fine with landing us landing this when the below is fixed and commits squashed. @tastejs/todomvc ?

- Do you really have to include these files? Seems more like dev dependencies.

```
examples/aurelia/jspm_packages/traceur.src.js
examples/aurelia/jspm_packages/es6-module-loader.src.js
examples/aurelia/jspm_packages/system.src.js
```

- Bug: Double-click to edit doesn't work when in a route (filter), like `Active`.

- Can you also fill out https://github.com/tastejs/todomvc/blob/master/learn.json for Aurelia

**Comment by mhoyer** [comment]
@sindresorhus, I was able to remove those ```.src.js``` files, fixed the double-click bug, filled the ```learn.json```, and I squashed everything to a single commit.

If any further support is needed, please let me know.

**Comment by addyosmani** [comment]
I'm :+1: on us landing this too. Confirmed the learn.json was populated as needed and the other bugs appear to have been addressed. Deferring to @sindresorhus for a merge when he's happy.

**Comment by sindresorhus** [comment]
Landed! Awesome work on getting this together @mhoyer :)

tumblr_mkwooez15j1qdlh1io1_400

**Comment by arthurvr** [comment]
W00t! Awesome work @mhoyer!

**Pull request closed without merging**